docs/adr/0023-localized-preferences.md

```diff
@@ -10,10 +10,30 @@ The problem is how to store the default values.
 
 ## Considered Options
 
-* _Localize Defaults_ (current implementation)
-* _Store the Preference only when it was changed by the user_. Otherwise, leave it empty. When a consumer gets such an empty preference, it knows that it needs to read the default and localize it. This won't work if users actually want something to be empty.
-* _Store the unlocalized String._ Consumers then check the String they got as a preference against the defaults. If it matches, localize it. Otherwise, use it.
+* Localized defaults
+* Store changed preferences
+* Store the unlocalized string
 
 ## Decision Outcome
 
-Chosen option: "_Store the unlocalized String._ Consumers then check the String they got as a preference against the defaults. If it matches, localize it. Otherwise, use it.", because Achieves goals without requiring too much refactoring and without (known) downsides.
+Chosen option: "Store the unlocalized string", because Achieves goals without requiring too much refactoring and without (known) downsides.
+
+## Pros and Cons of the Options
+
+### Localized defaults
+
+The default preference values are localized. That means that if JabRef does not find a value of the preference, it creates puts a default value. This value is the localized one.
+
+* Good, because This is the implementation as of August, 2021
+
+### Store changed preferences
+
+Store the preference only when it was changed by the user. Otherwise, leave it empty.
+
+* Good, because When a consumer gets such an empty preference, it knows that it needs to read the default and localize it.
+* Bad, because this won't work if users actually want something to be empty.
```

Comment:
Given the large amount of hacks / workarounds that are needed to make our localization tests work with the third approach, maybe this second approach is more efficient? One could use null / empty optional as the default value so that users can still provide an empty string if they wish.

Reply:
So the defaults hash-map would have an empy optional / null. When accessing we test against that. If it is not null/empty, we use that string without localization. If not, we localize a string. The question is, what string? Where do we store the actual default? Storing it where it is accessed is not a good idea IMO.

One solution would be to introduce a 'LocalizablePreference' class that has a boolean flag 'overwrittenByUser' and value. If the flag is set, we read the value from the preferences. If not, it stores the default value. Consumers can then decide whether to localize the value based on the flag.

@calixtus @koppor what do you think?

Reply:
One option would be to have the localized  default in the corresponding preference object.